File: api/agents.php
<?php

/**
 * Agents API Handler
 * Handles all agent-related API endpoints
 */

require_once __DIR__ . '/config.php';
require_once __DIR__ . '/db.php';

setCorsHeaders();

$method = $_SERVER['REQUEST_METHOD'];
$id = $_GET['id'] ?? null;

// Sample agents data
$sampleAgents = [
    [
        'id' => 1,
        'name' => 'John Mukama',
        'email' => 'john@edubridge.com',
        'phone' => '+256 700 123456',
        'region' => 'Kampala',
        'specialization' => 'Primary & Secondary Schools',
        'rating' => 4.8
    ],
    [
        'id' => 2,
        'name' => 'Sarah Nakato',
        'email' => 'sarah@edubridge.com',
        'phone' => '+256 700 234567',
        'region' => 'Jinja & Eastern Uganda',
        'specialization' => 'Islamic Schools',
        'rating' => 4.9
    ],
    [
        'id' => 3,
        'name' => 'David Okello',
        'email' => 'david@edubridge.com',
        'phone' => '+256 700 345678',
        'region' => 'Western Uganda',
        'specialization' => 'Technical Institutes',
        'rating' => 4.7
    ]
];

// Route requests
if ($method === 'GET' && $id) {
    getAgent($id);
} elseif ($method === 'GET') {
    getAgents();
} elseif ($method === 'POST') {
    registerAgent();
} else {
    jsonResponse(['error' => 'Method not allowed'], 405);
}

function getAgents()
{
    global $sampleAgents;

    $db = Database::getConnection();
    if ($db) {
        $agents = getAll('agents');
        if (count($agents) > 0) {
            jsonResponse([
                'success' => true,
                'agents' => $agents,
                'total' => count($agents)
            ]);
        }
    }

    jsonResponse([
        'success' => true,
        'agents' => $sampleAgents,
        'total' => count($sampleAgents)
    ]);
}

function getAgent($id)
{
    global $sampleAgents;

    foreach ($sampleAgents as $a) {
        if ($a['id'] == $id) {
            jsonResponse([
                'success' => true,
                'agent' => $a
            ]);
        }
    }

    $db = Database::getConnection();
    if ($db) {
        $agent = getById('agents', $id);
        if ($agent) {
            jsonResponse([
                'success' => true,
                'agent' => $agent
            ]);
        }
    }

    jsonResponse(['error' => 'Agent not found'], 404);
}

function registerAgent()
{
    $data = getJsonInput();

    if (empty($data['name']) || empty($data['email'])) {
        jsonResponse(['error' => 'Name and email are required'], 400);
    }

    $db = Database::getConnection();

    if ($db) {
        $data['created_at'] = date('Y-m-d H:i:s');
        $data['status'] = 'active';

        $id = insert('agents', $data);

        if ($id) {
            jsonResponse([
                'success' => true,
                'message' => 'Agent registered successfully',
                'id' => $id
            ], 201);
        }
    }

    jsonResponse([
        'success' => true,
        'message' => 'Agent registered successfully (demo mode)'
    ], 201);
}

File: api/agents.py
"""
Agents API Handler for Vercel
Handles all agent-related API endpoints
"""

import json

def get_response(status_code, body):
    return {
        "statusCode": status_code,
        "headers": {
            "Content-Type": "application/json",
            "Access-Control-Allow-Origin": "*",
            "Access-Control-Allow-Methods": "GET, POST, PUT, DELETE, OPTIONS",
            "Access-Control-Allow-Headers": "Content-Type, Authorization"
        },
        "body": json.dumps(body)
    }

# Sample agents data
SAMPLE_AGENTS = [
    {
        "id": 1,
        "name": "John Mukama",
        "email": "john@edubridge.com",
        "phone": "+256 700 123456",
        "region": "Kampala",
        "specialization": "Primary & Secondary Schools",
        "rating": 4.8
    },
    {
        "id": 2,
        "name": "Sarah Nakato",
        "email": "sarah@edubridge.com",
        "phone": "+256 700 234567",
        "region": "Jinja & Eastern Uganda",
        "specialization": "Islamic Schools",
        "rating": 4.9
    },
    {
        "id": 3,
        "name": "David Okello",
        "email": "david@edubridge.com",
        "phone": "+256 700 345678",
        "region": "Western Uganda",
        "specialization": "Technical Institutes",
        "rating": 4.7
    }
]

def handle(req, resource_id=None, params=None):
    method = req.get('method', 'GET')
    
    if method == 'OPTIONS':
        return get_response(200, {"message": "OK"})
    
    if method == 'GET' and resource_id is None:
        return get_response(200, {
            "success": True,
            "agents": SAMPLE_AGENTS,
            "total": len(SAMPLE_AGENTS)
        })
    
    if method == 'GET' and resource_id:
        agent = next((a for a in SAMPLE_AGENTS if str(a["id"]) == resource_id), None)
        if agent:
            return get_response(200, {"success": True, "agent": agent})
        return get_response(404, {"error": "Agent not found"})
    
    if method == 'POST':
        return get_response(201, {"success": True, "message": "Agent registered successfully"})
    
    return get_response(405, {"error": "Method not allowed"})

